src/Test/AtomicTests/objectives.jl

```diff
@@ -0,0 +1,105 @@
+#=
+    Functions in this file test functionality relating to objectives in MOI.
+
+### Requires
+    - optimize!
+
+### Functionality currently tested
+    - get/set ObjectiveSense
+    - a constant in a affine objective
+    - a blank objective
+
+### Functionality not yet tested
+    - Quadratic Objectives
+    - Modifications
+=#
+
+"""
+    Set objective to MaxSense
+"""
+function max_sense(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    @test MOI.isempty(model)
+    @test MOI.canset(model, MOI.ObjectiveSense())
+    MOI.set!(model, MOI.ObjectiveSense(), MOI.MaxSense)
+    @test MOI.canget(model, MOI.ObjectiveSense())
+    @test MOI.get(model, MOI.ObjectiveSense()) == MOI.MaxSense
+end
+atomictests["max_sense"] = max_sense
+
+"""
+    Set objective to MinSense
+"""
+function min_sense(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    @test MOI.isempty(model)
+    @test MOI.canset(model, MOI.ObjectiveSense())
+    MOI.set!(model, MOI.ObjectiveSense(), MOI.MinSense)
+    @test MOI.canget(model, MOI.ObjectiveSense())
+    @test MOI.get(model, MOI.ObjectiveSense()) == MOI.MinSense
+end
+atomictests["min_sense"] = min_sense
+
+"""
+    Set objective to FeasibilitySense
+"""
+function feasibility_sense(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    @test MOI.isempty(model)
+    @test MOI.canset(model, MOI.ObjectiveSense())
+    MOI.set!(model, MOI.ObjectiveSense(), MOI.FeasibilitySense)
+    @test MOI.canget(model, MOI.ObjectiveSense())
+    @test MOI.get(model, MOI.ObjectiveSense()) == MOI.FeasibilitySense
+end
+atomictests["feasibility_sense"] = feasibility_sense
+
+"""
+    Test constant in objective.
+"""
+function constant_obj(model::MOI.ModelLike, config::TestConfig)
```

Comment:
If the test calls `optimize!`, I'd propose to have "solve" somewhere in the name, like `solve_constant_obj`. You could easily think that `constant_obj` would just set and get the objective attribute without solving.